main/src/com/google/refine/exporters/sql/SqlExporter.java

```
@@ -0,0 +1,161 @@
+ */
+
+package com.google.refine.exporters.sql;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.refine.ProjectManager;
+import com.google.refine.browsing.Engine;
+import com.google.refine.exporters.CustomizableTabularExporterUtilities;
+import com.google.refine.exporters.TabularSerializer;
+import com.google.refine.exporters.WriterExporter;
+import com.google.refine.model.Project;
+import com.google.refine.util.JSONUtilities;
+
+public class SqlExporter implements WriterExporter {
+    
+    public static final String NO_COL_SELECTED_ERROR = "****NO COLUMNS SELECTED****";
+    public static final String NO_OPTIONS_PRESENT_ERROR = "****NO OPTIONS PRESENT****";
+
+    private final static Logger logger = LoggerFactory.getLogger("SqlExporter");
+
+    private List<String> columnNames = new ArrayList<String>();
+    private List<ArrayList<SqlData>> sqlDataList = new ArrayList<ArrayList<SqlData>>();
+    private JSONObject sqlOptions;
+
+    @Override
+    public String getContentType() {
+        return "text/plain";
+    }
+
+    @Override
+    public void export(final Project project, Properties params, Engine engine, final Writer writer)
+            throws IOException {
+        if(logger.isDebugEnabled()) {
+            logger.debug("export sql with params: {}", params);
+        }
+        TabularSerializer serializer = new TabularSerializer() {
+
+            @Override
+            public void startFile(JSONObject options) {
+                sqlOptions = options;
+                //logger.info("setting options::{}", sqlOptions);
+            }
+
+            @Override
+            public void endFile() {
+                try {
+                    if (columnNames.isEmpty()) {
+                        writer.write(NO_COL_SELECTED_ERROR);
```

Should this be controlled from the front end? We can have the validation at back end, but we may not need write the error into the dump file.

---

There is front end validation as well if you try to generate sql with no column selected. I can suppress the error dump but if this happens, the user will see an empty file.